tests/Simulation.py

```diff
@@ -0,0 +1,29 @@
+import pybop
+import pybamm
+import numpy as np
+
+# Form list of applied current
+measured_expirement = [np.arange(0,3,0.1),np.ones([30])]
+current_interpolant = pybamm.Interpolant(measured_expirement[0], measured_expirement[1], pybamm.t)
```

Comment:
Here is some sample usage with the current Simulation class, which is based on Pybamm.Simulation; however, it does not include the pybamm experiment methods as discussed above.

Question to the group, how far do we want to go in composing the pybamm methods? In this example, the only dependency on pybamm (at this level) is pybamm.Interpolant. Does it make sense for us to wrap the methods that are directly needed for pybop (creating drive cycles, updating parameters, plotting) and leave the rest for the user to acquire by importing pybamm?

Reply:
To expand and add a specific example, we would build a pybop.Interpolant class from the `pybamm.Interpolant` as,

```python
class Interpolant():
  def __init__(self):
    self.Interpolant= pybamm.Interpolant()
```

This provides the user with a unified package without having a required dependancy on pybamm. This is different to how pybamm-param has been constructed [example](https://github.com/paramm-team/pybamm-param/blob/develop/examples/scripts/mle.py). However, we would need to build classes to compose for all possible pybamm functionality.

Reply:
I wouldn't restrict yourself to reproducing pybamm's api, feel free to make your own api that is more suitable for parameterised models. Then you can just re-use whatever pybamm functionality you want inside that api.

E.g.

```python
# Create abstract model (no data or parameters thus far)
model = pybop.BaseSPM()

# Create a parameterisation object (now include parameters to be inferred, experimental conditions like input current, and link data to model outputs)
observation = pybop.Observed(""Current function [A]", np.linspace(0, 1, 10))
parameters = [
   pybop.Parameter("Electrode height [m]", prior=pybop.Normal(0, 1))
   pybop.Parameter("Negative particle radius [m]", prior=pybop.Uniform(0, 1))
   pybop.Parameter("Positive particle radius [m]", prior=pybop.Uniform(0, 10))
]
parameterisation = pybop.Parameterisation(model, observations = [observation], parameters=parameter)

# get MAP estimate, starting at a random initial point in parameter space
parameterisation.map(x0=[p.sample() for p in parameters]) 

# or sample from posterior
paramterisation.sample(1000, n_chains=4, ....)
```